### PR TITLE
fix(PN-9968): update session history removing replace() method

### PR DESCRIPTION
**Pull request by SarahDonvito**
## Short description
Update session history flow removing replace() method and using open() method with _self property in PF and PG

## How to test
Start PG/PF, click on "Accedi" button. When the page with SPID or CIE login is loaded, go back and check to be again in the page with only one "Accedi" button.

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import { vi } from 'vitest';
+
 import { EventPageType } from '@pagopa-pn/pn-commons';
 
 import { getConfiguration } from '../../services/configuration.service';
 import { getCurrentEventTypePage, goToLoginPortal } from '../navigation.utility';
 import { APP_STATUS, DELEGHE, DETTAGLIO_NOTIFICA, NOTIFICHE, RECAPITI } from '../routes.const';
 
 const replaceFn = vi.fn();
+window.open = vi.fn();
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts:10]: Instead to do this, copy the code from Footer.test.tsx. Do this for all tests below

File: packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { vi } from 'vitest';
+
 import { EventPageType } from '@pagopa-pn/pn-commons';
 
 import { getConfiguration } from '../../services/configuration.service';
 import { getCurrentEventTypePage, goToLoginPortal } from '../navigation.utility';
 import { APP_STATUS, DELEGHE, DETTAGLIO_NOTIFICA, NOTIFICHE, RECAPITI } from '../routes.const';
 
 const replaceFn = vi.fn();
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts:9]: If it is not used anymore, remove it. Do this for all tests below

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts (hunk 2 above)
Comment (SarahDonvito) [reply packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts:9]: What are you reffering to?

Comment (AndreaCimini90) [reply packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts]: to replaceFn

File: packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+const mockOpenFn = vi.fn();
 
 describe('Tests navigation utility methods', () => {
   const original = window.location;
 
   beforeAll(() => {
-    Object.defineProperty(window, 'location', {
-      writable: true,
-      value: { href: '', replace: replaceFn },
+    Object.defineProperty(window, 'open', {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/navigation.utility.test.ts:15]: rewrite the code in afterAll as for Footer.test.tsx. do it for all the test updated

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**